firestore/src/main/java/com/firebase/ui/firestore/paging/FirestorePagingOptions.java

```diff
@@ -0,0 +1,108 @@
+package com.firebase.ui.firestore.paging;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.paging.LivePagedListBuilder;
+import android.arch.paging.PagedList;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.util.DiffUtil;
+
+import com.firebase.ui.firestore.ClassSnapshotParser;
+import com.firebase.ui.firestore.SnapshotParser;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.Query;
+
+/**
+ * Options to conifigure an {@link FirestorePagingAdapter}.
+ */
+public class FirestorePagingOptions<T> {
+
+    private final PagingData mData;
+    private final SnapshotParser<T> mParser;
+    private final DiffUtil.ItemCallback<DocumentSnapshot> mDiffCallback;
+    private final LifecycleOwner mOwner;
+
+    private FirestorePagingOptions(@NonNull PagingData data,
+                                   @NonNull SnapshotParser<T> parser,
+                                   @NonNull DiffUtil.ItemCallback<DocumentSnapshot> diffCallback,
+                                   @Nullable LifecycleOwner owner) {
+        mData = data;
+        mParser = parser;
+        mDiffCallback = diffCallback;
+        mOwner = owner;
+    }
+
+    @NonNull
+    public PagingData getData() {
+        return mData;
+    }
+
+    @NonNull
+    public SnapshotParser<T> getParser() {
+        return mParser;
+    }
+
+    @NonNull
+    public DiffUtil.ItemCallback<DocumentSnapshot> getDiffCallback() {
+        return mDiffCallback;
+    }
+
+    @Nullable
+    public LifecycleOwner getOwner() {
+        return mOwner;
+    }
+
+    public static class Builder<T> {
+
+        private PagingData mData;
+        private SnapshotParser<T> mParser;
+        private LifecycleOwner mOwner;
+        private DiffUtil.ItemCallback<DocumentSnapshot> mDiffCallback;
+
+        @NonNull
+        public Builder<T> setQuery(@NonNull Query query,
+                                   @NonNull PagedList.Config config,
+                                   @NonNull Class<T> modelClass) {
+            return setQuery(query, config, new ClassSnapshotParser<T>(modelClass));
+        }
+
+        @NonNull
+        public Builder<T> setQuery(@NonNull Query query,
+                                   @NonNull PagedList.Config config,
+                                   @NonNull SnapshotParser<T> parser) {
+            // Build paged list
+            FirestoreDataSource.Factory factory = new FirestoreDataSource.Factory(query);
+            mData = new PagingData(new LivePagedListBuilder<>(factory, config).build());
+
+            mParser = parser;
+            return this;
+        }
+
+        @NonNull
+        public Builder<T> setLifecycleOwner(@NonNull LifecycleOwner owner) {
```

Comment:
Trying to keep the order consistent, this method should be below the diff callback one.